src/Server/MySQLHandler.cpp

```diff
@@ -530,16 +538,15 @@ void MySQLHandlerSSL::finishHandshakeSSL(
 
 static bool isFederatedServerSetupSetCommand(const String & query)
 {
-    static const std::regex expr{
+    static const re2::RE2 expr(
         "(^(SET NAMES(.*)))"
         "|(^(SET character_set_results(.*)))"
         "|(^(SET FOREIGN_KEY_CHECKS(.*)))"
         "|(^(SET AUTOCOMMIT(.*)))"
         "|(^(SET sql_mode(.*)))"
         "|(^(SET @@(.*)))"
-        "|(^(SET SESSION TRANSACTION ISOLATION LEVEL(.*)))"
-        , std::regex::icase};
```

Comment:
`std::regex::icase` -- Case insensitive.
you need to pass `re2::RE2::Options::set_case_sensitive(false)` to the `expr`.
As I see `case_sensitive` is `true` by default in re2.

Reply:
Correct, thanks, done.